spec/features/new_results_page_filters/vacancies_spec.rb

```diff
@@ -0,0 +1,51 @@
+require 'rails_helper'
+
+RSpec.feature 'Results page new vacancies filter' do
+  include StubbedRequests::Courses
+  include StubbedRequests::Subjects
+
+  let(:results_page) { PageObjects::Page::ResultsWithNewFilters.new }
+  let(:base_parameters) { results_page_parameters }
+
+  before do
+    activate_feature(:new_filters)
+
+    stub_subjects
+    stub_courses(query: base_parameters, course_count: 10)
+  end
+
+  describe 'viewing results without explicitly selecting a filter' do
+    it 'show courses with or without vacancies' do
+      results_page.load
+
+      expect(results_page.vacancies_filter.subheading.text).to eq('Vacancies')
+      expect(results_page.vacancies_filter.checkbox.checked?).to be(false)
+      expect(results_page.courses.count).to eq(10)
+    end
+  end
+
+  describe 'applying the filter' do
+    before do
+      stub_courses(
+        query: base_parameters.merge(
+          'filter[has_vacancies]' => 'true',
+          'filter[study_type]' => 'full_time,part_time',
+        ),
+        course_count: 10,
+      )
+    end
+
+    context 'show courses with vacancies only' do
+      it 'list the filtered courses' do
+        results_page.load
+
+        results_page.vacancies_filter.checkbox.check
+        results_page.apply_filters_button.click
+
+        expect(results_page.vacancies_filter.subheading.text).to eq('Vacancies')
+        expect(results_page.vacancies_filter.checkbox.checked?).to be(true)
+        expect(results_page.courses.count).to eq(10)
```

Comment:
I don't know if this is covered somewhere else and it would just be double testing, but should we stub 5 courses with no vacancies and 5 with and check that only 5 are returned when this is set to true?

Reply:
Yeah more legacy copy and paste.

Given we control the API response via course fixtures there is no added value in  asserting the number of courses displayed in these specs. I've removed